pkg/webhook/resources/storageclass/validator.go

```diff
@@ -38,13 +39,41 @@ func (v *storageClassValidator) Resource() types.Resource {
 }
 
 func (v *storageClassValidator) Create(_ *types.Request, newObj runtime.Object) error {
-	return v.validateSetUniqueDefault(newObj)
+	err := v.validateSetUniqueDefault(newObj)
+	if err != nil {
+		return err
+	}
+
+	return v.validateDataLocality(newObj)
 }
 
 func (v *storageClassValidator) Update(_ *types.Request, _ runtime.Object, newObj runtime.Object) error {
 	return v.validateSetUniqueDefault(newObj)
 }
 
+// Harvester rejects setting dataLocality as strict-local, because it makes volume non migrateable,
+// beside, strict-local volumes only could have one replica, it make Longhorn block node drain
+// https://longhorn.io/docs/1.5.3/references/settings/#node-drain-policy
+func (v *storageClassValidator) validateDataLocality(newObj runtime.Object) error {
+	sc := newObj.(*storagev1.StorageClass)
+	dataLocality, find := sc.Parameters[util.LonghornDataLocality]
+	if !find {
+		return nil
+	}
+
+	lhDataLocality := longhorn.DataLocality(dataLocality)
+
+	if lhDataLocality == longhorn.DataLocalityStrictLocal {
+		return werror.NewInvalidError("storage class with strict-local data locality is not allowed", sc.Name)
+	}
+
+	if lhDataLocality != longhorn.DataLocalityDisabled && lhDataLocality != longhorn.DataLocalityBestEffort {
+		return werror.NewInvalidError("storage class with invalid data locality", sc.Name)
```

Comment:
Also consider telling the user what to do (pseudo-code-y Sprintf)

```
"storage class with invalid data locality %q, valid values: %v",
    userSubmittedValue, strings.Join([]string{`"disabled"`,`"best-effort"`}, ", ")
```

Reply:
Updated per suggestion, thanks.